### PR TITLE
Extension to create EGLImage from QNX native buffer

### PR DESCRIPTION
**Pull request by stonesthrow**

### CODE REVIEW

File: extensions/QNX/EGL_QNX_image_native_buffer.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    EGL_KHR_image_base is required.
+
+    This extension is written against the wording of the EGL 1.4
+    Specification.
+
+Overview
+
+    This extension enables using an QNX buffer (struct
+    ??? _screen_buffer) as an EGLImage source.
```
Comment (stonesthrow) [inline extensions/QNX/EGL_QNX_image_native_buffer.txt:42]: What is the correct type name for qnx native buffer

File: extensions/QNX/EGL_QNX_image_native_buffer.txt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      |  <target>                  |  Notes                                  |
+      +----------------------------+-----------------------------------------+
+      |  EGL_NATIVE_BUFFER_QNX     |  Used for QNX buffer objects            |
+      +----------------------------+-----------------------------------------+
+       Table aaa.  Legal values for eglCreateImageKHR <target> parameter
+
+    ...
+
+    If <target> is EGL_NATIVE_BUFFER_QNX, <dpy> must be a valid display,
+    <ctx> must be EGL_NO_CONTEXT, <buffer> must be a valid ??? _screen_buffer
```
Comment (stonesthrow) [inline extensions/QNX/EGL_QNX_image_native_buffer.txt:76]: Need type for native buffer

File: extensions/QNX/EGL_QNX_image_native_buffer.txt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+     object (cast into the type EGLClientBuffer), and
+    attributes other than EGL_IMAGE_PRESERVED_KHR are ignored."
+
+    Add to the list of error conditions for eglCreateImageKHR:
+
+      "* If <target> is EGL_NATIVE_BUFFER_QNX and <buffer> is not a
+         valid ???, the error EGL_BAD_PARAMETER
```
Comment (stonesthrow) [inline extensions/QNX/EGL_QNX_image_native_buffer.txt:83]: Needs native buffer type - lets qnx developers know what to use

**Review by stonesthrow — COMMENTED** [review]
Open questions

File: extensions/QNX/EGL_QNX_image_native_buffer.txt (hunk 1 above)
Comment (stonesthrow) [reply extensions/QNX/EGL_QNX_image_native_buffer.txt:42]: screen_buffer_t

File: extensions/QNX/EGL_QNX_image_native_buffer.txt (hunk 3 above)
Comment (stonesthrow) [reply extensions/QNX/EGL_QNX_image_native_buffer.txt:83]: screen_buffer_t

**Comment by stonesthrow** [comment]
Needs update to egl.xml to assign token

**Comment by stonesthrow** [comment]
This comes after #180. There will be a possible conflict in egl.xml and registry.tcl

File: extensions/QNX/EGL_QNX_platform_screen.txt [hunk 4]
```diff
@@ -0,0 +1,89 @@
+Name
```
Comment (stonesthrow) [inline extensions/QNX/EGL_QNX_platform_screen.txt:1]: Something went sideways here. This should not be included.

**Review by stonesthrow — COMMENTED** [review]
eglext.h needs to be updated also.

File: extensions/QNX/EGL_QNX_platform_screen.txt (hunk 4 above)
Comment (stonesthrow) [reply extensions/QNX/EGL_QNX_platform_screen.txt]: Seems to have resolved when conflict resolved.

File: api/egl.xml [hunk 5]
```diff
@@ -3559,5 +3559,10 @@
                 <enum name="EGL_PLATFORM_SCREEN_QNX"/>
             </require>
         </extension>
+        <extension name="EGL_QNX_image_native_buffer" supported="egl">
+            <require>
+                <enum name="EGL_NATIVE_BUFFER_QNX"/>
```
Comment (oddhack) [inline api/egl.xml:3564]: Same issue here as the other extension - this just expresses a dependency on the enum, but there needs to be an actual definition of it with value 0x3551 in the corresponding \<enums> block above. Then I can generate the header update from egl.xml properly.

Comment (stonesthrow) [reply api/egl.xml:3565]: I see now. thanks. Then do I need to remove changes to eglext.h?

Comment (oddhack) [reply api/egl.xml:3565]: Preferably. I regenerate the headers when XML changes are merged - if people do it in their own branches it tends to lead to merge conflicts if several are in flight, as is happening right now.

**Comment by stonesthrow** [comment]
@oddhack Should be ready now for merge

**Pull request merged**